### ISSUE TITLE
Questions about online triplet loss

### ISSUE THREAD

**Issue by cri5Castro** [adambielski/siamese-triplet#45]
Hi there i'm using this repo (awesome job by the way) to pre-train a model  following the online triplet approach
Since i want to obtain embeddings of 512 dimensions instead of 2d embeddings, should I modify the pdist function? 
```
def pdist(vectors):
    distance_matrix = -2 * vectors.mm(torch.t(vectors)) + vectors.pow(2).sum(dim=1).view(1, -1) + vectors.pow(2).sum(
        dim=1).view(-1, 1)
    return distance_matrix
```

Sometimes my distance matrix has no 0's in the diagonal when i change the dimension to 512 , I wonder if  what change is needed maybe change it to something like this

```
def pdist(vectors):
    distance_matrix = -vectors.size()[1]* vectors.mm(torch.t(vectors)) + vectors.pow(2).sum(dim=1).view(1, -1) + vectors.pow(2).sum(
        dim=1).view(-1, 1)
    return distance_matrix
```
or should the loss the online triplet loss function with any embeddings dimension? by the way it is? 
thanks in advance
